### PR TITLE
use getter method for knative-build namespaces

### PR DESCRIPTION
**Pull request by zxDiscovery**
## Proposed Changes

Use getter method for `system.Namespace`. So that we can create a shared library for this in knative/pkg to standardize this across Knative. https://github.com/knative/serving/pull/2708

### CODE REVIEW

**Comment by zxDiscovery** [comment]
The PR https://github.com/knative/serving/pull/2708 has already merged. Is this PR ready to merge? @ImJasonH

File: pkg/system/names.go [hunk 1]
```diff
@@ -16,17 +16,33 @@ limitations under the License.
...
+	NamespaceEnvKey = "SYSTEM_NAMESPACE"
 )
 
-func init() {
-	Namespace = os.Getenv("SYSTEM_NAMESPACE")
-	if Namespace == "" {
-		Namespace = "knative-build"
+// Namespace holds the K8s namespace where our build system
+// components run.
+func Namespace() string {
```
Comment (mattmoor) [inline pkg/system/names.go]: Can we hoist this package into `knative/pkg` first, and apply the rest of this update when picking that up?  You are going to have to touch all of these files for both anyways.

Comment (zxDiscovery) [reply pkg/system/names.go]: https://github.com/knative/pkg/pull/235. Is this right?

**Comment by zxDiscovery** [comment]
https://github.com/knative/pkg/pull/235 has been merged. Can we merge this PR now? @mattmoor

**Comment by mattmoor** [comment]
@zxDiscovery this doesn't use the copy of `system` from `knative/pkg`?

**Comment by zxDiscovery** [comment]
> @zxDiscovery this doesn't use the copy of `system` from `knative/pkg`?

Done @mattmoor

**Pull request merged**